tests/unit/python/foglamp/common/test_statistics.py

```diff
@@ -24,92 +23,108 @@
         storageMock1 = MagicMock(spec=StorageClient)
-        s1 = Statistics(storageMock1)
+        s1 = await statistics.create_statistics(storageMock1)
         storageMock2 = MagicMock(spec=StorageClient)
-        s2 = Statistics(storageMock2)
+        s2 = await statistics.create_statistics(storageMock2)
         assert s1._storage == s2._storage
 
-    def test_register(self):
+    async def test_register(self):
         """ Test that register results in a database insert """
         storageMock = MagicMock(spec=StorageClient)
-        stats = Statistics(storageMock)
-        loop = asyncio.get_event_loop()
-        loop.run_until_complete(stats.register('T1Stat', 'Test stat'))
-        args, kwargs = stats._storage.insert_into_tbl.call_args
-        assert args[0] == 'statistics'
-        expected_storage_args = json.loads(args[1])
-        assert expected_storage_args['key'] == 'T1Stat'
-        assert expected_storage_args['value'] == 0
-        assert expected_storage_args['previous_value'] == 0
-        assert expected_storage_args['description'] == 'Test stat'
+        stats = statistics.Statistics(storageMock)
+        stats._registered_keys = []
+        async def mock_coro():
+            return {"response": "updated", "rows_affected": 1}
+
+        with patch.object(stats._storage, 'insert_into_tbl', return_value=mock_coro()) as stat_update:
+            await stats.register('T1Stat', 'Test stat')
+            args, kwargs = stat_update.call_args
+            assert args[0] == 'statistics'
+            expected_storage_args = json.loads(args[1])
+            assert expected_storage_args['key'] == 'T1Stat'
+            assert expected_storage_args['value'] == 0
+            assert expected_storage_args['previous_value'] == 0
+            assert expected_storage_args['description'] == 'Test stat'
         stats._storage.insert_into_tbl.reset_mock()
 
-    def test_register_twice(self):
+    async def test_register_twice(self):
         """ Test that register results in a database insert only once for same key"""
         storageMock = MagicMock(spec=StorageClient)
-        stats = Statistics(storageMock)
-        loop = asyncio.get_event_loop()
-        loop.run_until_complete(stats.register('T2Stat', 'Test stat'))
-        loop.run_until_complete(stats.register('T2Stat', 'Test stat'))
-        assert stats._storage.insert_into_tbl.called
-        assert stats._storage.insert_into_tbl.call_count == 1
+        stats = statistics.Statistics(storageMock)
+        stats._registered_keys = []
+        async def mock_coro():
+            return {"response": "updated", "rows_affected": 1}
+
+        with patch.object(stats._storage, 'insert_into_tbl', return_value=mock_coro()) as stat_update:
+            await stats.register('T2Stat', 'Test stat')
+            await stats.register('T2Stat', 'Test stat')
+            assert stat_update.called
+            assert stat_update.call_count == 1
```

Comment:
Fix the indentation of stat_update assertions. Also better to assert its payload too and stat_update var should be stat_insert or patch_insert_tbl

Reply:
payload verification with register is already done in test_register. The purpose of this test is to verify that with the same key, insert_into_tbl is called just once. Other concerns fixed.